engine/cdn/cdn_gc.go

```diff
@@ -14,41 +19,76 @@ const (
 	ItemLogGC = 24 * 3600
 )
 
-func (s *Service) CompleteWaitingItems(ctx context.Context) {
-	tick := time.NewTicker(1 * time.Minute)
-	defer tick.Stop()
+// ItemsGC clean long incoming item + delete item from buffer when synchronized everywhere
+func (s *Service) ItemsGC(ctx context.Context) {
+	tickGC := time.NewTicker(1 * time.Minute)
+	defer tickGC.Stop()
 	for {
 		select {
 		case <-ctx.Done():
 			if ctx.Err() != nil {
 				log.Error(ctx, "cdn:CompleteWaitingItems: %v", ctx.Err())
 			}
 			return
-		case <-tick.C:
-			itemUnits, err := storage.LoadOldItemUnitByItemStatusAndDuration(ctx, s.Mapper, s.mustDBWithCtx(ctx), index.StatusItemIncoming, ItemLogGC)
-			if err != nil {
-				log.Warning(ctx, "cdn:CompleteWaitingItems: unable to get items ids: %v", err)
-				continue
+		case <-tickGC.C:
+			if err := s.cleanBuffer(ctx); err != nil {
+				log.ErrorWithFields(ctx, logrus.Fields{"stack_trace": fmt.Sprintf("%+v", err)}, "%s", err)
 			}
-			log.Debug("cdn:CompleteWaitingItems: %d items to complete", len(itemUnits))
-			for _, itemUnit := range itemUnits {
-				tx, err := s.mustDBWithCtx(ctx).Begin()
-				if err != nil {
-					log.Error(ctx, "cdn:CompleteWaitingItems: unable to start transaction: %v", err)
-					continue
-				}
-				if err := s.completeItem(ctx, tx, itemUnit); err != nil {
-					_ = tx.Rollback()
-					log.Warning(ctx, "cdn:CompleteWaitingItems: unable to complete item %s: %v", itemUnit.ItemID, err)
-					continue
-				}
-				telemetry.Record(ctx, metricsItemCompletedByGC, 1)
-				if err := tx.Commit(); err != nil {
-					_ = tx.Rollback()
-					log.Warning(ctx, "cdn:CompleteWaitingItems: unable to commit transaction: %v", err)
-					continue
-				}
+			if err := s.cleanWaitingItem(ctx); err != nil {
+				log.ErrorWithFields(ctx, logrus.Fields{"stack_trace": fmt.Sprintf("%+v", err)}, "%s", err)
 			}
 		}
 	}
 }
+
+func (s *Service) cleanBuffer(ctx context.Context) error {
+	var cdsBackendID string
+	for _, sto := range s.Units.Storages {
+		_, ok := sto.(*cds.CDS)
+		if !ok {
+			continue
+		}
+		cdsBackendID = sto.ID()
+		break
+	}
+	if cdsBackendID == "" {
+		return nil
+	}
+	itemIDs, err := storage.LoadAllItemsIDInBufferAndAllUnitsExceptCDS(s.mustDBWithCtx(ctx), cdsBackendID)
```

Comment:
Why do weed wait for sync in all backend instead of just one that is not the buffer or CDS ?